test/metric/metric_value_query.go

```diff
@@ -52,17 +58,34 @@ type MetricValueFetcher interface {
 	// getPluginSupportedMetric returns the supported metrics for each plugin
 	// https://github.com/aws/amazon-cloudwatch-agent/blob/6451e8b913bcf9892f2cead08e335c913c690e6d/translator/translate/metrics/config/registered_metrics.go
 	getPluginSupportedMetric() map[string]struct{}
+	setEnv(env *environment.MetaData)
 }
 
-type baseMetricValueFetcher struct{}
+type baseMetricValueFetcher struct {
+	Env *environment.MetaData
+}
 
-func (f *baseMetricValueFetcher) fetch(namespace, metricName string, metricSpecificDimensions []types.Dimension, stat Statistics) (MetricValues, error) {
+func (f *baseMetricValueFetcher) getEnv() *environment.MetaData {
+	return f.Env
+}
+
+func (f *baseMetricValueFetcher) setEnv(env *environment.MetaData) {
+	f.Env = env
+}
```

Comment:
you should be able to just do `fetcher.Env = f.Env` above. I don't think that we need the getter/setter model here.

Reply:
where? I tried different things to get Env from all the fetchers, and making getEnv() was the only way. direct access to variable failed which is.. I guess reasonable

Reply:
TBH, I don't like `getter/setter` model since its too hollow for me and does not give any value much. IIRC you can access variable as long  as its uppercase when outside of function.

Reply:
fetch logic changed https://github.com/aws/amazon-cloudwatch-agent-test/pull/26